src/encode/json.rs

```diff
@@ -163,6 +168,49 @@ impl Deserialize for JsonEncoderDeserializer {
     }
 }
 
+#[cfg(feature = "log_kv")]
+mod kv {
+    use log::kv::VisitSource;
+
+    pub fn get_attributes(source: &dyn log::kv::Source) -> anyhow::Result<Vec<(String, String)>> {
+        Ok(LogKvVisitor::process(source)?.yield_values())
+    }
+    #[derive(Default)]
+    struct LogKvVisitor {
```

Comment:
Why a custom struct vs BTreeMap like the example in log::kv?

Reply:
I am a bit on the fence about this. My initial reasoning is that a Vec is more memory efficient, which enables us to minimize overall footprint of the system. However, looking at the `kv` documentation there is an explicit mention of the fact that
`A source doesn’t have to guarantee any ordering or uniqueness of key-values. If the given visitor returns an error then the source may early-return with it, even if there are more key-values.`.
This means that we need to guarantee uniqueness in order to produce valid JSON (`serde_json` also doesn't enforce uniqueness AFAIK).
So I think you're right and I'll switch it to a BTreeMap.